xbmc/video/guilib/VideoStreamSelectHelper.cpp

```diff
@@ -0,0 +1,478 @@
+    }
+    return infoA.samplerate < infoB.samplerate;
+  }
+};
+
+struct SortComparerStreamSubtitle
+{
+  bool operator()(const std::pair<int, SubtitleStreamInfoExt>& a,
+                  const std::pair<int, SubtitleStreamInfoExt>& b)
+  {
+    const SubtitleStreamInfoExt& infoA = a.second;
+    const SubtitleStreamInfoExt& infoB = b.second;
+
+    if (infoA.languageDesc != infoB.languageDesc)
+    {
+      return infoA.languageDesc < infoB.languageDesc;
+    }
+    if (infoA.isDefault != infoB.isDefault)
+    {
+      return infoA.isDefault < infoB.isDefault;
+    }
+    if (infoA.isForced != infoB.isForced)
+    {
+      return infoA.isForced < infoB.isForced;
+    }
+    if (infoA.isHearingImpaired != infoB.isHearingImpaired)
+    {
+      return infoA.isHearingImpaired < infoB.isHearingImpaired;
+    }
+    if (infoA.isVisualImpaired != infoB.isVisualImpaired)
+    {
+      return infoA.isVisualImpaired < infoB.isVisualImpaired;
+    }
+    return infoA.codecName < infoB.codecName;
+  }
+};
+
+} // unnamed namespace
+
+void KODI::VIDEO::GUILIB::OpenDialogSelectVideoStream()
+{
+  CGUIDialogSelect* dialog{CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogSelect>(
+      WINDOW_DIALOG_SELECT_VIDEO_STREAM)};
+  if (!dialog)
+  {
+    CLog::LogF(LOGERROR, "Unable to get WINDOW_DIALOG_SELECT_VIDEO_STREAM dialog instance");
+    return;
+  }
+
+  auto& components = CServiceBroker::GetAppComponents();
+  auto appPlayer = components.GetComponent<CApplicationPlayer>();
+  const int streamCount = appPlayer->GetVideoStreamCount();
+  const int selectedIndex = appPlayer->GetVideoStream();
+
+  std::vector<std::pair<int, VideoStreamInfoExt>> streams;
+  streams.reserve(streamCount);
+
+  // Collect all streams
+  for (int i = 0; i < streamCount; ++i)
+  {
+    VideoStreamInfoExt info;
+    appPlayer->GetVideoStreamInfo(i, info);
+
+    info.isDefault = info.flags & StreamFlags::FLAG_DEFAULT;
+    info.isForced = info.flags & StreamFlags::FLAG_FORCED;
+    info.isHearingImpaired = info.flags & StreamFlags::FLAG_HEARING_IMPAIRED;
+    info.isVisualImpaired = info.flags & StreamFlags::FLAG_VISUAL_IMPAIRED;
+
+    streams.emplace_back(i, info);
+  }
+
+  // Sort streams
+  std::sort(streams.begin(), streams.end(), SortComparerStreamVideo());
+
+  // Convert streams to FileItem's
+  CFileItemList itemsToDisplay;
+  itemsToDisplay.Reserve(streams.size());
+
+  for (auto& [index, info] : streams)
+  {
+    const auto fileItem = std::make_shared<CFileItem>(info.name);
+    fileItem->SetProperty("stream.index", index);
+    fileItem->SetProperty("stream.description", info.name);
+    fileItem->SetProperty("stream.codec", info.codecName);
+
+    std::string languageDesc;
+    g_LangCodeExpander.Lookup(info.language, languageDesc);
+    fileItem->SetProperty("stream.language", languageDesc);
+
+    fileItem->SetProperty("stream.resolution",
+                          std::to_string(info.width) + "x" + std::to_string(info.height));
```

Comment:
This is a good example for problems with hardcoding strings that should be localizable. The lower case letter x is not the best choice here, there is a unicode value that fits better, at least for my language. => U+2715 (10005) "times"

Reply:
you cant use U+2715 since its not always present on the font sets,
also the standard "x" char is used everywhere on kodi as on resolution lists, system info etc...

Reply:
> also the standard "x" char is used everywhere on kodi as on resolution lists, system info etc...

That does not make it right. 

It’s a pity and not a serious issue in this case, but localization seems to be something not many of us want to spend time for improvements. 

Anyways, nothing we should consider a blocker for this PR.

Reply:
anyway it would still be a half done thing, should be found a solution for all use cases
the problems are also on skin side, e.g.
```
<label>$INFO[Player.Process(videowidth),[COLOR button_focus]$LOCALIZE[38031]:[/COLOR] ,x]$INFO[Player.Process(videoheight)]$INFO[Player.Process(videoscantype)]$INFO[Player.Process(videodar),$COMMA , AR]$INFO[Player.Process(videofps),$COMMA , FPS]</label>
```
these are not localizable

should be provided a way to allow skins to apply localizations on id's that use formatting placeholders, e.g.
i can create something like e.g.
```
msgctxt "#50000"
msgid "{0:d}x{1:d}"
msgstr ""
```
but if you use $LOCALIZE[50000] how you can send the Player.Process(videowidth) Player.Process(videoheight) values from xml?
maybe extending LOCALIZE to accept parameters
$LOCALIZE[50000, Player.Process(videowidth), Player.Process(videoheight), ... ]
that will format the string id with provided infolabels, the bigger problem are the value type conversions from string (CGUIInfoManager/infolabels seem to convert all to strings)